### PR TITLE
refactor: derive thiserror for store error

### PR DESCRIPTION
**Pull request by KaoImin**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,6 +23,7 @@ serde_json = "1.0"
 log = "0.4"
 env_logger = "0.8"
 tokio = { version = "1.4", features = ["macros", "rt-multi-thread"] }
+derive_more = "0.99"
```
Comment (driftluo) [inline Cargo.toml]: Anyhow’s error already has `Display`, why should `derive_more` be introduced?

Comment (KaoImin) [reply Cargo.toml]: Anyhow requires error implied `std::error::Error` which needs to derive `Display` trait firstly.

Comment (driftluo) [reply Cargo.toml]: I thought anyhow exported the derive Error macro of thiserror, but not :(

**Comment by doitian** [comment]
Why?

I think anyhow is for the errors that we cannot (or don't want to) handle programmatically and just print it in log or show it to users.

**Comment by doitian** [comment]
I recommend to use thiserror to implement `std::Error` only in this repo.

**Comment by KaoImin** [comment]
> Why?
> 
> I think anyhow is for the errors that we cannot (or don't want to) handle programmatically and just print it in log or show it to users.

In ckb-companion, we use ckb-indexer as a library. It is This is convenient to convert error by `?` directly.

**Comment by doitian** [comment]
> > Why?
> > I think anyhow is for the errors that we cannot (or don't want to) handle programmatically and just print it in log or show it to users.
> 
> In ckb-companion, we use ckb-indexer as a library. It is This is convenient to convert error by `?` directly.

Anyhow supports `From<impl std::Error>`, thus it's enough to implement std Error for ckb indexer errors.

**Review by doitian — APPROVED** [review]

File: src/service.rs [hunk 2]
```diff
@@ -156,6 +156,9 @@ pub trait CkbRpc {
 
     #[rpc(name = "local_node_info")]
     fn local_node_info(&self) -> Result<LocalNode>;
+
+    #[rpc(name = "get_tip_header")]
+    fn get_tip_header(&self) -> Result<Option<HeaderView>>;
```
Comment (keroro520) [inline src/service.rs]: When `get_tip_header` returns `None` ?

**Pull request merged**